src/modules/roles/controllers/role.controller.ts

```diff
@@ -1,119 +1,88 @@
+import { getRolesResponse, createRoleResponse, updateRolesBulkResponse, deleteRoleResponse, getRoleCountResponse, getRoleResponse, updateRoleResponse } from '../docs/role-response.docs.js'
+import { UuidParam } from '../../../utils/params/uuid-param.utiil.js'
 
 @ApiTags('Roles')
 @Controller('roles')
-@ApiOAuth2([])
+@ApiBearerAuth()
 export class RoleController {
   constructor (
     private readonly roleService: RoleService
   ) {}
 
-  @Get()
-  @ApiResponse({
-    status: 200,
-    description: 'The roles has been successfully received.',
-    type: [RoleTransformerType]
-  })
+  @Get('/')
+  @ApiResponse(getRolesResponse)
   @Permissions(Permission.ROLE_READ)
-  async getRoles (): Promise<RoleTransformerType[]> {
+  public async getRoles (): Promise<RoleTransformerType[]> {
     const roles = await this.roleService.findAll()
-
     return new RoleTransformer().array(roles)
   }
 
-  @Post()
-  @ApiResponse({
-    status: 201,
-    description: 'The role has been successfully created.',
-    type: RoleTransformerType
-  })
+  @Post('/')
+  @ApiResponse(createRoleResponse)
   @Permissions(Permission.ROLE_CREATE)
-  async createRole (
+  public async createRole (
     @Body() createRoleDto: CreateRoleDto
   ): Promise<RoleTransformerType> {
     const role = await this.roleService.create(createRoleDto)
-
     return new RoleTransformer().item(role)
   }
 
-  @Post('bulk')
-  @ApiResponse({
-    status: 201,
-    description: 'The roles has been successfully created.',
-    type: [RoleTransformerType]
-  })
+  @Post('/bulk')
+  @ApiResponse(updateRolesBulkResponse)
   @Permissions(Permission.ROLE_UPDATE)
-  async updateRolesBulk (
+  public async updateRolesBulk (
     @Body() updateRolesBulk: UpdateRolesBulkDto
   ): Promise<RoleTransformerType[]> {
     const roles = await this.roleService.updateBulk(updateRolesBulk)
-
     return new RoleTransformer().array(roles)
   }
 
-  @Get(':role')
-  @ApiResponse({
-    status: 200,
-    description: 'The role has been successfully received.',
-    type: RoleTransformerType
-  })
+  @Get('/:roleUuid')
+  @ApiResponse(getRoleResponse)
   @Permissions(Permission.ROLE_READ)
-  async getRole (
-    @Param('role', ParseUUIDPipe) uuid: string
+  public async getRole (
+    @UuidParam('roleUuid') roleUuid: string
   ): Promise<RoleTransformerType> {
-    const role = await this.roleService.findOne(uuid)
-
+    const role = await this.roleService.findOne(roleUuid)
     return new RoleTransformer().item(role)
   }
 
-  @Post(':role')
-  @ApiResponse({
-    status: 200,
-    description: 'The role has been successfully updated.',
-    type: RoleTransformerType
-  })
+  @Post('/:roleUuid')
+  @ApiResponse(updateRoleResponse)
   @Permissions(Permission.ROLE_UPDATE)
-  async updateRole (
-    @Body() updateRoleDto: CreateRoleDto,
-    @Param('role', ParseUUIDPipe) uuid: string
+  public async updateRole (
+    @UuidParam('roleUuid') roleUuid: string,
+    @Body() updateRoleDto: CreateRoleDto
```

Comment:
```suggestion
    @Body() updateRoleDto: UpdateRoleDto
```
Aparte type voor update?